src/config/reader.rs

```diff
@@ -82,8 +82,11 @@ impl ConfigReader {
             let new_config = self.read_script(new_config).await?;
             config = config.merge_right(&new_config);
         }
-
-        Ok(config)
+        let config_set = ConfigSet::from(config);
+        let config_set = config_set
+            .resolve_extensions(self.file.clone(), self.http.clone())
+            .await;
```

Comment:
Have an instance of `ConfigSetResolver` in `ConfigReader` and use that resolver to create a config_set.